File: Data Structures & Algorithms/two-integer-sum/submission-11.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        // Dictionary method
        // Key, value = value, index
        Dictionary<Integer, Integer> d = new Hashtable<>();
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int current_num = nums[i];
            int complement = target - current_num;

            if (((Hashtable<Integer, Integer>) d).containsKey(complement)) {
                int[] return_this = new int[] { d.get(complement), i };
                return return_this;
            }

            d.put(current_num, i);
        }

        return new int[] { 0, 0 };
    }
}

File: Data Structures & Algorithms/is-subsequence/submission-11.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        l, r = 0, 0
        n = len(t)

        while r < n and l < len(s):
            if s[l] == t[r]:
                l+=1
            r+=1

        return l == len(s)